harness/determined/profiler.py

```diff
@@ -760,6 +763,7 @@ def run(self) -> None:
                     for msr in self.accumulating_measurements.values():
                         self.metrics_batch.append(msr.metric_type, msr.metric_name, msr)
                     self.accumulating_measurements = {}
+                    break
```

Comment:
why this break? we may not necessarily have timed out yet, right?

Reply:
Maybe I misunderstood the former code.  This is not intended to force a flush then?

Reply:
fixed.